src/mongocxx/test/hint.cpp

```diff
@@ -52,6 +52,22 @@ TEST_CASE("Hint", "[hint]") {
             auto index_doc = builder::stream::document{} << "a" << 1 << builder::stream::finalize;
             REQUIRE(index_hint != index_doc);
         }
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
```

Comment:
let's keep the tests and remove MONGOCXX_DEPRECATED and have a ticket for doing that later after we figure out how to suppress deprecation warnings in testing.